mapper/orm/src/main/java/org/hibernate/search/mapper/orm/outbox/impl/OutboxEvent.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.orm.outbox.impl;
+
+import java.util.Arrays;
+import java.util.Objects;
+import javax.persistence.Transient;
+
+import org.hibernate.search.mapper.pojo.route.DocumentRoutesDescriptor;
+import org.hibernate.search.util.common.serialization.spi.SerializationUtils;
+
+public final class OutboxEvent {
+
+	public enum Type {
+		ADD, ADD_OR_UPDATE, DELETE
+	}
+
+	private Integer id;
+	private String entityName;
+	private String serializedId;
+	private byte[] serializedRoutes;
+	private Type type;
+
+	@Transient
+	private Object identifier;
```

Comment:
I missed that in my original review, but can we make sure the name of these properties are straightforward?

* There's no point saying the values are "serialized", as that's mostly a matter of encoding. We don't say `integerCount` for a property representing a count that happens to be encoded as an integer, so there's no reason to say `serializedId` for a property reprensenting an ID that happens to be encoded as a serialized string.
* Since there's an even id and an entity ID, we should probably make it clear that a given property is about the entity, and another about the even.

So maybe something like this?

```suggestion
	private Integer id;
	private Type type;
	
	private String entityName;
	private String entityId;

	private byte[] documentRoutes;

	@Transient
	private Object originalEntityId;
```